### PR TITLE
hiIQ and rewards stuff

### PR DESCRIPTION
**Pull request by FortisFortuna**

### CODE REVIEW

File: everipediaiq/everipediaiq.abi [hunk 1]
```diff
@@ -375,67 +303,57 @@
         {
             "name": "burn",
             "type": "burn",
-            "ricardian_contract": "Burn IQ tokens and take them out of circulation."
+            "ricardian_contract": ""
```
Comment (kesar) [inline everipediaiq/everipediaiq.abi:306]: ricardian contract texts have been removed. perhaps worth to rollback

```suggestion
            "ricardian_contract": "Burn IQ tokens and take them out of circulation."
```

File: eparticlectr/eparticlectr.contracts.md [hunk 2]
```diff
@@ -10,47 +8,27 @@ Purge pre-EPID reward orphans
...
 <h1 class="contract">brainmeart</h1>
 Stake IQ (only callable by eparticlectr).
-<h1 class="contract">finalize</h1>
-Finalize a proposal and calculate rewards & slashings.
 <h1 class="contract">finalizeextr</h1>
 Finalize a proposal and calculate rewards & slashings. New Version
-<h1 class="contract">slashnotify</h1>
-Logging action to notify users of slashes.
 <h1 class="contract">slashnotifex</h1>
 Logging action to notify users of slashes. New Version
```
Comment (kesar) [inline eparticlectr/eparticlectr.contracts.md:14]: this action is removed 
```suggestion
```

File: eparticlectr/eparticlectr.abi [hunk 3]
```diff
@@ -968,135 +698,60 @@
...
-        {
-            "name": "brainclmid",
-            "type": "brainclmid",
-            "ricardian_contract": "Claim back IQ that was staked for Brainpower. Must happen after the 21 day staking period."
-        },
         {
             "name": "brainclmidex",
             "type": "brainclmidex",
-            "ricardian_contract": "Claim back IQ that was staked for Brainpower. Must happen after the 21 day staking period. New Version"
+            "ricardian_contract": ""
```
Comment (kesar) [inline eparticlectr/eparticlectr.abi:704]: ricardian contracts are removed
```suggestion
            "ricardian_contract": "Claim back IQ that was staked for Brainpower. Must happen after the 21 day staking period. New Version"
```

File: eparticlectr/eparticlectr.cpp [hunk 4]
```diff
@@ -357,76 +255,27 @@ void eparticlectr::finalizeextr( uint64_t proposal_id ) {
                 });
             }
         }
-        // Reduce staking time for vote winners to 5 days
-        else if (approved && !vote_it->is_editor) {
```
Comment (kesar) [inline eparticlectr/eparticlectr.cpp:361]: why remove the approved checking here? its used in the first if but not in this one

File: eparticlectr/eparticlectr.cpp [hunk 5]
```diff
@@ -357,76 +255,27 @@ void eparticlectr::finalizeextr( uint64_t proposal_id ) {
...
-                });
-            }
+                if (vote_it->approve == approved){
+                    // Winners
+                    staketable.modify( stake_it, same_payer, [&]( auto& a ) {
+                        a.completion_time = a.timestamp + WINNING_VOTE_STAKE_TIME;
+                    });
+                }
+                else {
+                    // Losers
```
Comment (kesar) [inline eparticlectr/eparticlectr.cpp:270]: should here get the extra slashed time? extraSecsSlash?

File: eparticlectr/eparticlectr.cpp [hunk 6]
```diff
@@ -357,76 +255,27 @@ void eparticlectr::finalizeextr( uint64_t proposal_id ) {
                 });
             }
         }
-        // Reduce staking time for vote winners to 5 days
-        else if (approved && !vote_it->is_editor) {
+        // Reduce staking time for vote winners to 5 days. Keep losers the same
+        else if (!vote_it->is_editor) {
```
Comment (kesar) [inline eparticlectr/eparticlectr.cpp:259]: I wonder if we should keep here the approved or not. the if still has the approved condition, not sure if there can be some race condition where this approved can guard

File: eparticlectr/eparticlectr.cpp (hunk 5 above)
Comment (FortisFortuna) [reply eparticlectr/eparticlectr.cpp:270]: I think we decided to stop slashing

File: eparticlectr/eparticlectr.cpp (hunk 6 above)
Comment (FortisFortuna) [reply eparticlectr/eparticlectr.cpp:259]: I will fix the logic

Comment (FortisFortuna) [reply eparticlectr/eparticlectr.cpp:259]: actually, the `if (vote_it->approve == approved)` logic below handles it

**Pull request merged**